PlayLoop/PlayerInteractionHandler.js

```diff
@@ -0,0 +1,201 @@
+            this.blockers.splice(index, 1);
+        }
+    }
+
+    static isTargeting(obj) {
+        return this.currentTarget === obj;
+    }
+
+    static canTrigger(obj) {
+        return (this.isTargeting(obj) && this.blockers.length === 0);
+    }
+
+    static willAllowJump() {
+        let target = this.currentTarget || this.lastTarget;
+
+        if (!target) {
+            return true;
+        }
+
+        if (target.type === ObjectTypes.NPC) {
+            return false;
+        } 
+        if (target.type === ObjectTypes.DOOR) {
+            return (target.activationTrigger !== DOOR_TRIGGERS.UP);
+        }
+    }
+
+    static updateControl(id) {
+        // should probably merge updateControl stuff into Controller and read from there instead?
+        let buttonDown = Controller[id];
+        this.justPressed[id] = buttonDown && !this.pressed[id];
+        this.pressed[id] = buttonDown;
+    }
+
+    static handleInput() {
+        // should probably merge updateControl stuff into Controller and read from there instead?
+        // controls need to be updated every frame
+        this.updateControl('jump');
+        this.updateControl('up');
+        this.updateControl('down');
+
+        if (!this.currentTarget) {
+            return;
+        }
+        if (this.blockers.length > 0) {
+            this.recentlyBlocked = true;
+            return;
+        }
+
+        if (this.recentlyBlocked) {
+            // wait an extra frame before interacting if recently blocked
+            this.recentlyBlocked = false;
+            return;
+        }
+
+        if (this.currentTarget.type === ObjectTypes.NPC) {
+            if (this.justPressed.jump) {
+                this.currentTarget.startDialogue();
+            }
+        }
+        if (this.currentTarget.type === ObjectTypes.DOOR) {
+            let triggerType = this.currentTarget.activationTrigger;
+            let cueEntrance = false;
+
+            if (triggerType === DOOR_TRIGGERS.AUTOMATIC) {
+                cueEntrance = true;
+            }
+            else if (triggerType === DOOR_TRIGGERS.UP) {
+                cueEntrance = this.justPressed.up;
+            }
+            else if (triggerType === DOOR_TRIGGERS.DOWN) {
+                cueEntrance = this.justPressed.down;
+            }
+            
+            if (cueEntrance) {
+                this.currentTarget.sendPlayerThroughExit();
+            }
+        }
+    }
+
+    static updatePlayerTarget(player) {
+        if (player.death) {
+            this.lastTarget = this.currentTarget;
+            this.currentTarget = null;
+            return;
+        }
+
+        let playerX = player.x;
+        let playerY = player.y;
+        let minDist = Infinity;
+        let nearestTarget = null;
+
+        this.registeredTargets.forEach((target) => {
+            let xDist = target.x - playerX;
+            let yDist = target.y - playerY;
+            let dist = Math.sqrt((xDist * xDist) + (yDist * yDist));
```

Comment:
Is this something you'd like to introduce as part of a "style guide" for the repo? I didn't notice much of a pattern in the codebase so far, but could adopt it going forward. 
From a personal standpoint, I feel like I've annoyed myself _just_ frequently enough with presumptuous `const`-ing (where later things change and it turns out I need a `let` instead) that I tend to use it pretty sparingly. I've also seen suggestions to avoid `const` strictly out of people being confused about the particulars of it (allowing mutability of a referenced object), but haven't been bit by that so far.